src/huggingface_hub/hf_api.py

```diff
@@ -1964,6 +1960,18 @@ def create_commit(
                 If the Hub API returns an HTTP 400 error (bad request)
             :class:`ValueError`:
                 If `create_pr` is `True` and revision is neither `None` nor `"main"`.
+            :class: `RepositoryNotFoundError`:
+                If repository is not found (error 404): wrong repo_id/repo_type, private
+                but not authenticated or repo does not exist.
+
+        <Tip warning={true}>
+
+        `create_commit` assumes that the repo already exists on the Hub. If you get a
+        Client error 404, please make sure you are authenticated and that `repo_id` and
+        `repo_type` are set correctly. If repo does not exist, create it first using
+        [`~huggingface_hub.hf_api.create_repo`].
```

Comment:
```suggestion
        [`~hf_api.create_repo`].
```
Nit: `huggingface_hub` is optional (but if you prefer to have it, feel free to ignore this).